tests/integrations/admin/indexation-integration-test.php

```diff
@@ -0,0 +1,367 @@
+
+		Monkey\Functions\expect( 'esc_html__' )
+			->andReturnFirstArg();
+
+		Monkey\Functions\expect( '__' )
+			->andReturnFirstArg();
+
+		// Mock WP nonce.
+		Monkey\Functions\expect( 'wp_create_nonce' )
+			->once()
+			->andReturn( 'nonce' );
+
+		$expected_data = [
+			'amount'  => 40,
+			'ids'     => [
+				'count'    => '#yoast-indexation-current-count',
+				'progress' => '#yoast-indexation-progress-bar',
+			],
+			'restApi' => [
+				'root'      => 'https://example.org/wp-ajax/',
+				'endpoints' =>
+					[
+						'posts'    => 'yoast/v1/indexation/posts',
+						'terms'    => 'yoast/v1/indexation/terms',
+						'archives' => 'yoast/v1/indexation/post-type-archives',
+						'general'  => 'yoast/v1/indexation/general',
+					],
+				'nonce'     => 'nonce',
+			],
+			'message' => [
+				'indexingCompleted' => '<span class="wpseo-checkmark-ok-icon"></span>Good job! All your site\'s content has been indexed.',
+				'indexingFailed'    => 'Something went wrong indexing the content of your site. Please try again later.',
+			],
+			'l10n'    => [
+				'calculationInProgress' => 'Calculation in progress...',
+				'calculationCompleted'  => 'Calculation completed.',
+				'calculationFailed'     => 'Calculation failed, please try again later.',
+			],
+		];
+
+		// The script should be localized with the right data.
+		Monkey\Functions\expect( 'wp_localize_script' )
+			->with(
+				WPSEO_Admin_Asset_Manager::PREFIX . 'indexation',
+				'yoastIndexationData',
+				$expected_data
+			);
+
+		$this->instance->enqueue_scripts();
+	}
+
+	/**
+	 * Tests that scripts and styles are not enqueued when there is
+	 * nothing to index.
+	 *
+	 * @covers ::enqueue_scripts
+	 * @covers ::get_total_unindexed
+	 */
+	public function test_enqueue_scripts_when_nothing_should_be_indexed() {
+		// Nothing should be indexed.
+		$this->set_total_unindexed_expectations(
+			[
+				'post_type_archive' => 0,
+				'general'           => 0,
+				'post'              => 0,
+				'term'              => 0,
+			]
+		);
+
+		// The warning and modal should not be rendered.
+		Monkey\Actions\expectAdded( 'admin_footer' )->never();
+		Monkey\Actions\expectAdded( 'admin_notices' )->never();
+
+		// The script should not be localized.
+		Monkey\Functions\expect( 'wp_localize_script' )->never();
+
+		$this->instance->enqueue_scripts();
+	}
+
+	/**
+	 * Tests that the indexation warning is shown when its respective method is called.
+	 *
+	 * @covers ::render_indexation_warning
+	 */
+	public function test_render_indexation_warning() {
+		// Mock i18n and string escape functions.
+		Monkey\Functions\expect( 'esc_html__' )
+			->andReturnFirstArg();
+
+		Monkey\Functions\expect( '__' )
+			->andReturnFirstArg();
+
+		Monkey\Functions\expect( 'esc_js' )
+			->andReturnFirstArg();
+
+		Monkey\Functions\expect( 'esc_attr__' )
+			->andReturnFirstArg();
+
+		// Mock WP nonce.
+		Monkey\Functions\expect( 'wp_create_nonce' )
+			->once()
+			->andReturn( 'nonce' );
```

Comment:
I think most of these would do better living in the `TestCase` as default stubs. In fact, `__` and `esc_js` are already there. Maybe `wp_create_nonce` is the exception here since, maybe not.

Meaning you don't have to specify all this in every test, so please review more of these in your other test if you decide to change this 🙂